util/dnsname/dnsname.go

```diff
@@ -17,3 +111,13 @@ func HasSuffix(name, suffix string) bool {
 	nameBase := strings.TrimSuffix(name, suffix)
 	return len(nameBase) < len(name) && strings.HasSuffix(nameBase, ".")
 }
+
+// TrimSuffix trims any trailing dots from a name and removes the
+// suffix ending if present.
```

Comment:
further clarify the returned name will never end in a period, even after trimming?